spanner/api/Spanner/Program.cs

```diff
@@ -1757,35 +1757,44 @@ await connection.BeginTransactionAsync())
             string connectionString =
                 $"Data Source=projects/{projectId}/instances/{instanceId}"
                 + $"/databases/{databaseId}";
-            List<Singer> singers = new List<Singer> {
-                new Singer {SingerId = firstSingerId, FirstName = "Marc",
-                    LastName = "Richards"},
-                new Singer {SingerId = secondSingerId, FirstName = "Catalina",
-                    LastName = "Smith"},
-                new Singer {SingerId = 3, FirstName = "Alice",
-                    LastName = "Trentor"},
-                new Singer {SingerId = 4, FirstName = "Lea",
-                    LastName = "Martin"},
-                new Singer {SingerId = 5, FirstName = "David",
-                    LastName = "Lomond"},
+            List<Album> albums = new List<Album> {
+                new Album { SingerId = secondSingerId, AlbumId = 1, AlbumTitle = "Green" },
+                new Album { SingerId = secondSingerId, AlbumId = 3, AlbumTitle = "Terrified" },
             };
             // Create connection to Cloud Spanner.
             using (var connection = new SpannerConnection(connectionString))
             {
                 await connection.OpenAsync();
 
-                // Delete rows from the Singers table.
-                var cmd = connection.CreateDeleteCommand("Singers",
+                // Delete individual rows from the Albums table.
+                var cmd = connection.CreateDeleteCommand(
+                    "Albums",
                     new SpannerParameterCollection {
-                        {"SingerId", SpannerDbType.Int64}
-                    });
-                await Task.WhenAll(singers.Select(singer =>
+                        {"SingerId", SpannerDbType.Int64},
+                        {"AlbumId", SpannerDbType.Int64}
+                    }
+                );
+                await Task.WhenAll(albums.Select(album =>
                 {
-                    cmd.Parameters["SingerId"].Value = singer.SingerId;
+                    cmd.Parameters["SingerId"].Value = album.SingerId;
```

Comment:
This won't work when executing several commands in parallel as you are doing here. The Parameters collection will be the same for all executions so you are overriding the parameters values every time. You'll end up deleting only a few of the rows, those for which the values were copied to the request before the "next value change" happened.
You should either execute the commands sequentially or create a new command for each delete, i.e. add the command creation here.

Reply:
Oh interesting, I didn't realize and for some reason, it didn't come up with the tests. Thanks for spotting this.

Reply:
Since you are opening the connection yourself the first await actually happens when you send the request, so this very particular example was not a problem. But, if you weren't opening the connection then the first await would happen before sending the request, to open the connection, or if you were doing this inside threads (we had an issue exactly because of that the other day), then it will be a problem. It's best to always show the prefered approach, which is to have individual parameter collections or commands, instead of explaining all this nuances I think.